packages/relay/src/lib/eth.ts

```diff
@@ -1574,27 +1578,35 @@ export class EthImpl implements Eth {
    * @param returnLatest
    */
   private async getHistoricalBlockResponse(blockNumberOrTag?: string | null, returnLatest?: boolean, requestId?: string | undefined): Promise<any | null> {
-    let blockResponse: any;
-    // Determine if the latest block should be returned and if not then just return null
+    const maxBlockRange = Number(process.env.MAX_BLOCK_RANGE) || 5;
     if (!returnLatest && EthImpl.blockTagIsLatestOrPending(blockNumberOrTag)) {
       return null;
     }
-
+  
+    if (blockNumberOrTag != null && blockNumberOrTag.length < 32 && !EthImpl.blockTagIsEarliestOrPending(blockNumberOrTag)) {
```

Comment:
I think we've introduced dupklication here.
If `blockNumberOrTag` === `latest` will you not end up making the following calls twice? Here and then on line 1595
```
const latestBlockResponse = await this.mirrorNodeClient.getLatestBlock(requestId);
      const latestBlock = latestBlockResponse.blocks[0];
```

Reply:
You're right.  The fix is easy since we're only interested in actual block numbers, so we can add a !EthImpl.blockTagIsLatestOrPending(blockNumberOrTag) to that condition.